src/pages/[platform]/build-a-backend/add-aws-services/analytics/set-up-analytics/index.mdx

```diff
@@ -0,0 +1,513 @@
+
+<Callout>
+
+visionOS support is currently in **preview** and can be used by targeting the [`visionos-preview`](https://github.com/aws-amplify/amplify-swift/tree/visionos-preview) branch. 
+As new Xcode 15 beta versions are released, the branch will be updated with any necessary fixes on a best effort basis.
+
+For more information on how to use the `visionos-preview` branch, see [Platform Support](https://github.com/aws-amplify/amplify-swift/tree/visionos-preview#platform-support).
+
+</Callout>
+</InlineFilter>
+
+<InlineFilter filters={["android"]}>
+## Prerequisites
+
+* An Android application targeting Android API level 16 (Android 4.1) or above
+
+</InlineFilter>
+
+<InlineFilter filters={["flutter"]}>
+
+- A Flutter application targeting Flutter SDK >=3.3.0 with Amplify libraries integrated
+
+  The following are also required, depending on which platforms you are targeting:
+
+  - An iOS configuration targeting at least iOS 13.0 and XCode version >=13.2
+  - An Android configuration targeting at least Android API level 24 (Android 7.0) or above
+  - Any browser supported by Flutter for Web (you can check the list of supported browsers [here](https://docs.flutter.dev/development/platform-integration/web/faq#which-web-browsers-are-supported-by-flutter))
+  - Any Windows OS meeting Flutter minimums
+  - macOS version 10.15 or higher
+  - Any Ubuntu Linux distribution meeting Flutter minimums
+
+
+</InlineFilter>
+## Set up Analytics backend
+
+The following is an example utilizing the [AWS Cloud Development Kit (AWS CDK)](https://docs.aws.amazon.com/cdk/latest/guide/home.html) to create the an Analytics resource powered by [Amazon Pinpoint](https://aws.amazon.com/pinpoint/). 
+
+```ts title="amplify/backend.ts"
+import { Policy, PolicyStatement } from "aws-cdk-lib/aws-iam";
+import { CfnApp } from "aws-cdk-lib/aws-pinpoint";
+import { Stack } from "aws-cdk-lib";
+
+const backend = defineBackend({
+  auth, 
+  data,
+  // additional resources 
+});
+
+const analyticsStack = backend.createStack("analytics-stack");
+
+// create a Pinpoint app
+const pinpoint = new CfnApp(analyticsStack, "Pinpoint", {
+  name: "myPinpointApp",
+});
+
+// create an IAM policy to allow interacting with Pinpoint
+const pinpointPolicy = new Policy(analyticsStack, "PinpointPolicy", {
+  policyName: "PinpointPolicy",
+  statements: [
+    new PolicyStatement({
+      actions: ["mobiletargeting:UpdateEndpoint", "mobiletargeting:PutEvents"],
+      resources: [pinpoint.attrArn + "/*"],
+    }),
+  ],
+});
+
+// apply the policy to the authenticated and unauthenticated roles
+backend.auth.resources.authenticatedUserIamRole.attachInlinePolicy(pinpointPolicy);
+backend.auth.resources.unauthenticatedUserIamRole.attachInlinePolicy(pinpointPolicy);
+
+// patch the custom Pinpoint resource to the expected output configuration
+backend.addOutput({
+  analytics: {
+    pinpoint_app_id: pinpoint.ref,
+    aws_region: Stack.of(pinpoint).region,
+  },
+});
```

Comment:
```suggestion
backend.addOutput({
  analytics: {
    amazon_pinpoint: {
      app_id: pinpoint.ref,
      aws_region: Stack.of(pinpoint).region,
    }
  },
});
```

Reply:
<img width="1840" alt="image" src="https://github.com/aws-amplify/docs/assets/4989523/98048a24-8d46-4032-bfe2-e165db9febef">
when I make `record` call, it triggers this error

Reply:
Turns out there's a bug in the workfrlow. A temp workaround is:
```

Amplify.configure(amplifyOutputs)
Amplify.configure({
  ...Amplify.getConfig(),
  Analytics: amplifyOutputs.Analytics
})

```

Reply:
updated the configure example